apps/enterprise-knowledge-retrieval/enterprise_knowledge_retrieval.ipynb

```diff
@@ -0,0 +1,1750 @@
+    "REDIS_DB = '0'\n",
+    "\n",
+    "redis_client = r(host=REDIS_HOST, port=REDIS_PORT, db=REDIS_DB,decode_responses=False)\n",
+    "\n",
+    "\n",
+    "# Constants\n",
+    "VECTOR_DIM = 1536 # length of the vectors\n",
+    "PREFIX = \"wiki\" # prefix for the document keys\n",
+    "DISTANCE_METRIC = \"COSINE\" # distance metric for the vectors (ex. COSINE, IP, L2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 104,
+   "id": "4cb5247d",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 104,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Create search index\n",
+    "\n",
+    "# Index\n",
+    "INDEX_NAME = \"wiki-index\"           # name of the search index\n",
+    "VECTOR_FIELD_NAME = 'content_vector'\n",
+    "\n",
+    "# Define RediSearch fields for each of the columns in the dataset\n",
+    "# This is where you should add any additional metadata you want to capture\n",
+    "id = TextField(\"id\")\n",
+    "url = TextField(\"url\")\n",
+    "title = TextField(\"title\")\n",
+    "text_chunk = TextField(\"content\")\n",
+    "file_chunk_index = NumericField(\"file_chunk_index\")\n",
+    "\n",
+    "# define RediSearch vector fields to use HNSW index\n",
+    "\n",
+    "text_embedding = VectorField(VECTOR_FIELD_NAME,\n",
+    "    \"HNSW\", {\n",
+    "        \"TYPE\": \"FLOAT32\",\n",
+    "        \"DIM\": VECTOR_DIM,\n",
+    "        \"DISTANCE_METRIC\": DISTANCE_METRIC\n",
+    "    }\n",
+    ")\n",
+    "# Add all our field objects to a list to be created as an index\n",
+    "fields = [url,title,text_chunk,file_chunk_index,text_embedding]\n",
+    "\n",
+    "redis_client.ping()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 126,
+   "id": "266b8aee",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 126,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "redis_client.flushall()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "33c07f00",
+   "metadata": {},
+   "source": [
+    "Optional step to drop the index if it already exists\n",
+    "\n",
+    "```redis_client.ft(INDEX_NAME).dropindex()```\n",
+    "\n",
+    "If you want to clear the whole DB use:\n",
+    "\n",
+    "```redis_client.flushall()```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 127,
+   "id": "08f30b56",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Unknown Index name\n",
+      "Not there yet. Creating\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Check if index exists\n",
+    "try:\n",
+    "    redis_client.ft(INDEX_NAME).info()\n",
+    "    print(\"Index already exists\")\n",
+    "except Exception as e:\n",
```

Comment:
best practice is to catch specific exceptions rather than naked exceptions. But in this case it looks like the error is a ResponseError from redis.exceptions.ResponseError, so maybe not worth it in this case.